### PR TITLE
FISH-9771 FISH-9688 spot bugs random only used once

### PR DESCRIPTION
**Pull request by NotedSalmon**
Used `SecurityUtils` to generate a random and use it within the respective classes. Spot bugs issue. (check description)
## Description
Spotbugs warning:
```
Random object created and used only once in com.sun.enterprise.admin.cli.cluster.ImportSyncBundleCommand.backupInstanceDir()

This code creates a java.util.Random object, uses it to generate one random number, and then discards the Random object. This produces mediocre quality random numbers and is inefficient. If possible, rewrite the code so that the Random object is created once and saved, and each time a new random number is required invoke a method on the existing Random object to obtain it.

If it is important that the generated Random numbers not be guessable, you must not create a new Random for each random number; the values are too easily guessable. You should strongly consider using a java.security.SecureRandom instead (and avoid allocating a new SecureRandom for each random number needed).random objects.
```
### Testing Performed

built payara and ran spotbugs against the changes made.

### CODE REVIEW

File: nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java [hunk 1]
```diff
@@ -93,6 +93,8 @@
 @ReferenceConstraint(skipDuringCreation=true, payload=Cluster.class)
 public interface Cluster extends ConfigBeanProxy, PropertyBag, Named, SystemPropertyBag, ReferenceContainer, RefContainer, Payload {
 
+    SecureRandom R = new SecureRandom();
```
Comment (svendiedrichsen) [inline nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java:96]: Pretty thrifty name for it.

File: nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java [hunk 2]
```diff
@@ -132,6 +132,7 @@ public class ImportSyncBundleCommand extends LocalInstanceCommand {
     private File backupDir;
 
     private static final String RENDEZVOUS_PROPERTY_NAME = "rendezvousOccurred";
+    private static final SecureRandom r = new SecureRandom();
```
Comment (svendiedrichsen) [inline nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java:135]: Thrifty as well.

File: nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java (hunk 1 above)
Comment (pzygielo) [reply nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java:96]: It is static already.

**Review by aubi — APPROVED** [review]
LGTM

File: nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java [hunk 3]
```diff
@@ -614,6 +615,7 @@ class Decorator implements CreationDecorator<Cluster> {
         public void decorate(AdminCommandContext context, final Cluster instance) throws TransactionFailure, PropertyVetoException {
             Logger logger = ConfigApiLoggerInfo.getLogger();
             LocalStringManagerImpl localStrings = new LocalStringManagerImpl(Cluster.class);
+            Random random = new SecureRandom();
```
Comment (svendiedrichsen) [inline nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java:618]: * Is it nessessary to use a SecureRandom for this? SecureRandom might block if the underlying system runs out of random byte input.
* wouldn't it be sufficient to have a static Random member

File: nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java [hunk 4]
```diff
@@ -132,6 +132,7 @@ public class ImportSyncBundleCommand extends LocalInstanceCommand {
     private File backupDir;
 
     private static final String RENDEZVOUS_PROPERTY_NAME = "rendezvousOccurred";
+    private static final Random random = new SecureRandom();
```
Comment (svendiedrichsen) [inline nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java:135]: Is it nessessary to use a SecureRandom for this?

File: nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java (hunk 3 above)
Comment (pzygielo) [inline nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java:618]: 1. Now `random` gets created even if it's not going to be used.
2. I can not see `random` to be used more than once. Both references seem to be in exclusive branches.

File: nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java (hunk 4 above)
Comment (aubi) [reply nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java:135]: "usually static properties are written in capital letters"

https://www.oracle.com/technetwork/java/codeconventions-150003.pdf
I suppose you mean "Constant", e.g. chapter 9. Random is not a constant, although it is static and final.

File: nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java (hunk 3 above)
Comment (aubi) [reply nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java:618]: You are right, just `Random` is really much faster (I tested via JMH).
This place is executed during cluster-related commands, e.g. once only. So for this usage, I suggest to leave the original SecureRandom.

File: nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java (hunk 4 above)
Comment (aubi) [reply nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ImportSyncBundleCommand.java:135]: ad SecureRandom -- see my comment up

File: nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java (hunk 3 above)
Comment (aubi) [reply nucleus/admin/config-api/src/main/java/com/sun/enterprise/config/serverbeans/Cluster.java]: The new solution solves this issue -- the static field in a different class is created only if it is used.
Ad used more than once: it's what spotbugs reported, in fact it is called only once, of course.

**Review by aubi — APPROVED** [review]
LGTM

File: nucleus/common/common-util/src/main/java/org/glassfish/common/util/RandomUtils.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.security.SecureRandom;
+
+/**
+ * Centralize usage of random number generator.
+ *
+ * @author: Petr Aubrecht
+ */
+public class RandomUtils {
+    private static SecureRandom random = new SecureRandom();
```
Comment (svendiedrichsen) [inline nucleus/common/common-util/src/main/java/org/glassfish/common/util/RandomUtils.java:50]: Can and should be final, shouldn't it?

File: nucleus/common/common-util/src/main/java/com/sun/enterprise/universal/security/SecurityUtils.java [hunk 6]
```diff
@@ -48,9 +48,9 @@
  * @author Byron Nevins
  */
 public final class SecurityUtils {
+    private static SecureRandom random = new SecureRandom();
 
     public static String getSecureRandomHexString(int numBytes) {
-        SecureRandom random = new SecureRandom();
         byte[] bb = new byte[numBytes];
         random.nextBytes(bb);
```
Comment (jGauravGupta) [inline nucleus/common/common-util/src/main/java/com/sun/enterprise/universal/security/SecurityUtils.java:55]: Use `RandomUtils` here to consolidate the `SecureRandom` usage into a single utility class to maintain consistency and manage randomness more effectively.

**Review by jGauravGupta — CHANGES_REQUESTED** [review]

File: nucleus/common/common-util/src/main/java/com/sun/enterprise/universal/security/SecurityUtils.java (hunk 6 above)
Comment (jGauravGupta) [reply nucleus/common/common-util/src/main/java/com/sun/enterprise/universal/security/SecurityUtils.java:55]: Should be fixed separately as may cause OSGI issue, due to package naming.

**Review by jGauravGupta — APPROVED** [review]

**Pull request merged**